roles/cluster_infra/defaults/main.yml

```diff
@@ -50,3 +50,12 @@ cluster_groups_validation:
 cluster_groups_zenith:
   # Any hosts in the grafana and openondemand groups should go in the zenith group
   zenith: [grafana, openondemand]
+
+# Deploy user SSH keys
+cluster_deploy_ssh_keys: >-
+  {{
+     cluster_deploy_ssh_additional_public_keys | default([]) +
+     ( [cluster_deploy_ssh_public_key]
+     if cluster_deploy_ssh_public_key is defined
+     else [] )
+  }}
```

Comment:
Maybe it would be less complicated if we just have an extra variable, `cluster_additional_ssh_public_keys: []`, and continue using the existing variables?

E.g.:

```
    ssh_authorized_keys:
    {%- if cluster_user_ssh_public_key is defined %}
      - {{ cluster_user_ssh_public_key }}
    {%- endif %}
    {%- if cluster_deploy_ssh_public_key is defined %}
      - {{ cluster_deploy_ssh_public_key }}
    {%- else %}
      - "${openstack_compute_keypair_v2.cluster_keypair.public_key}"
    {%- endif %}
    {%- for ssh_key in cluster_additional_ssh_public_keys %}
      - {{ ssh_key }}
    {%- endfor %}
```

Reply:
Yes  - I wanted to generate the list elsewhere because it made the whitespace formatting in the template much easier, but that's fixed with `#jinja2: trim_blocks:False`.

Now using this approach.